### ISSUE TITLE
Increase code coverage

### ISSUE THREAD

**Issue by moreal** [zeromq/netmq#849]
Now we need to increase the coverage...

_Originally posted by @somdoron in https://github.com/zeromq/netmq/pull/848#issuecomment-578403651_

I agree for it. In my thought, the *codecov* should be added into required checks (e.g., appveyor build, CodeFactor, ...) to stop the decreasement of the coverage.

**Comment by somdoron** [comment]
Sounds good to me

**Comment by moreal** [comment]
Then could you add it after observe other opinions for times? Because I am not maintainer and have no permissions for this repository.
IMHO: It seems not sure to collect enough opinions to say that the opinions was came and gone, from other users. Or can it be decided only with maintainers' confirms?

**Comment by moreal** [comment]
Yeah, I expect it will help stay code stable 😄. Is this needed more discussion? or it's ready to apply?

**Comment by moreal** [comment]
It seems already applied. Is it better to close this? Or is it better to leave it open for reminding? Tell me your opinions.

**Comment by somdoron** [comment]
Lets leave it open. If you want to change anything at the project level let
me know and I would give you a permission

**Comment by moreal** [comment]
I'm sorry to delete the comment because it was not really applied yet. Now there are some checks like below.

![image](https://user-images.githubusercontent.com/26626194/73958340-24620580-494b-11ea-9d51-f3a9d38bb685.png)

But I also want to add *codecov* as checks like below. because if not, it can be merged without check ing code coverage of it.

![image](https://user-images.githubusercontent.com/26626194/73960390-27122a00-494e-11ea-899d-391e43e99b23.png)

Further more, I want to suggest to set up an issue management strategy and pre-release. Also I'm sorry but I couldn't understand completely what the *permission* means there.. Could you let me know more about it please?

**Comment by moreal** [comment]
If there was some parts I didn't understood well, let me know that 😁 @somdoron

**Comment by somdoron** [comment]
@moreal I'm not sure what I need to do in order to add the codecov, can you link to a tutorial or something?

**Comment by moreal** [comment]
There is helpful document provided from codecov, [here][codecov-docs-link]. I'm not sure but it might require to install [codecov github app] on this repository! If there was hard to set up and the document is not enough, I'll try to write simple tutorial with gist or screen capture. Or it's just thought, if you can give me a temporary permission, I could try it.

[codecov github app]: https://github.com/apps/codecov/
[codecov-docs-link]: https://docs.codecov.io/docs/commit-status

**Comment by somdoron** [comment]
>  if you can give me a temporary permission, I could try it.

that is what I suggest earlier.

**Comment by somdoron** [comment]
you have access.

**Comment by moreal** [comment]
> > if you can give me a temporary permission, I could try it.
> 
> that is what I suggest earlier.

I have understood.

> you have access.

Then that means you gave me access permission already? If not, let's set the date and time you gave access!

**Comment by somdoron** [comment]
I gave you a permission

**Comment by moreal** [comment]
Ok, I saw and got it. I sent request to install codecov github app, please check email or notification about it. And I couldn't access to *Settings -> Branches -> Branch protection rules* even I got the permission. It might the permission not enough to access there.

**Comment by moreal** [comment]
I open pull request to resolve the problem. please review it.

**Comment by moreal** [comment]
I let this issue stay opened to encourage contributors to increase code coverage with test code when writing pull request. So this issue should not be closed and is not stale.
